### PR TITLE
Feature/changes video mapping refactor

### PR DESCRIPTION
**Pull request by salogranada**
Lorem Ipsum

### CODE REVIEW

File: include/cv_camera/capture.hpp [hunk 1]
```diff
@@ -0,0 +1,278 @@
+// Copyright [2015] Takashi Ogura<t.ogura@gmail.com>
```
Comment (pepisg) [inline include/cv_camera/capture.hpp]: Why did you change header files' extension? this makes harder to know what we changed with respect to the original repo. If there is not a good reason to do this I would suggest keeping the original `.h` files

File: src/driver.cpp [hunk 2]
```diff
@@ -1,113 +1,175 @@
...
-  camera_->setPropertyFromParam(CV_CAP_PROP_SATURATION, "cv_cap_prop_saturation");
-  camera_->setPropertyFromParam(CV_CAP_PROP_HUE, "cv_cap_prop_hue");
-  camera_->setPropertyFromParam(CV_CAP_PROP_GAIN, "cv_cap_prop_gain");
-  camera_->setPropertyFromParam(CV_CAP_PROP_EXPOSURE, "cv_cap_prop_exposure");
-  camera_->setPropertyFromParam(CV_CAP_PROP_CONVERT_RGB, "cv_cap_prop_convert_rgb");
-
-  camera_->setPropertyFromParam(CV_CAP_PROP_RECTIFICATION, "cv_cap_prop_rectification");
-  camera_->setPropertyFromParam(CV_CAP_PROP_ISO_SPEED, "cv_cap_prop_iso_speed");
+
+    camera_->setPropertyFromParam(cv::CAP_PROP_POS_MSEC, "cv_cap_prop_pos_msec");
```
Comment (pepisg) [inline src/driver.cpp]: there are several changes that look like they were caused by the autoformatter. Is very hard to review the PR this way beause we don't know what changes affect the implementation and what are only style. We should keep only the changes that concern the implementation

Comment (salogranada) [reply src/driver.cpp]: Done

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,53 +1,91 @@
...
-find_package(catkin REQUIRED COMPONENTS
-  image_transport roscpp cv_bridge sensor_msgs nodelet camera_info_manager roslint)
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
```
Comment (garzonwilmer5) [inline CMakeLists.txt]: Why not C++17?

File: include/cv_camera/capture.h [hunk 4]
```diff
@@ -47,16 +52,32 @@ class Capture
...
-  void open(const std::string &device_path);
+  bool open(const std::string &port);
 
+  /**
+   * @brief Finds the equivalent camera device associated to
+   *        one port.
+   * @param command Terminal command to be executed
+   * @return command_output::Command output delivered by the terminal
+   */
+  std::string execute_command(const char* command);
```
Comment (garzonwilmer5) [inline include/cv_camera/capture.h:81]: There is already a function for that, `executeShellCommand` from utils/console.hpp

File: include/cv_camera/capture.h [hunk 5]
```diff
@@ -156,12 +190,49 @@ class Capture
   /**
    * @brief rescale camera calibration to another resolution
    */
-  void rescaleCameraInfo(int width, int height);
+  void rescaleCameraInfo(uint width, uint height);
```
Comment (garzonwilmer5) [inline include/cv_camera/capture.h:227]: Why did you put the function body in the header file?

File: include/cv_camera/capture.h [hunk 6]
```diff
@@ -156,12 +190,49 @@ class Capture
...
+              return "rgba8";
+          default:
+              throw std::runtime_error("Unsupported encoding type");
+      }
+  }
+
+  /**
+   * @brief set current time to message header
+   */
+  void set_now(builtin_interfaces::msg::Time& time)
```
Comment (garzonwilmer5) [inline include/cv_camera/capture.h]: Same here

File: src/capture.cpp [hunk 7]
```diff
@@ -44,146 +58,220 @@ void Capture::loadCameraInfo()
...
+    RCLCPP_ERROR(node_->get_logger(), "Unable to open device /dev/video%d.", device_id);
+    return false;
   }
-  pub_ = it_.advertiseCamera(topic_name_, buffer_size_);
-
+  else
+  {
+    RCLCPP_WARN_ONCE(node_->get_logger(), "Opening topic %s: with /dev/video%d", img_topic_name_.c_str(), device_id);
+  }
+  // pub_ = it_.advertiseCamera(topic_name_, buffer_size_);
```
Comment (garzonwilmer5) [inline src/capture.cpp]: Remove death code

File: src/cv_camera_node.cpp [hunk 8]
```diff
@@ -4,24 +4,14 @@
...
-    }
-  }
-  catch (cv_camera::DeviceError &e)
-  {
-    ROS_ERROR_STREAM("cv camera open failed: " << e.what());
-    return 1;
-  }
-
+  rclcpp::init(argc, argv);
+  rclcpp::executors::SingleThreadedExecutor exec;
```
Comment (garzonwilmer5) [inline src/cv_camera_node.cpp]: Consider using an [StaticSingleThreadedExecutor](https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1executors_1_1StaticSingleThreadedExecutor.html)

**Review by garzonwilmer5 — CHANGES_REQUESTED** [review]

File: include/cv_camera/capture.h (hunk 5 above)
Comment (salogranada) [reply include/cv_camera/capture.h:227]: Missed that, done

File: include/cv_camera/capture.h (hunk 6 above)
Comment (salogranada) [reply include/cv_camera/capture.h]: Done

File: src/capture.cpp (hunk 7 above)
Comment (salogranada) [reply src/capture.cpp]: Done

File: src/cv_camera_node.cpp (hunk 8 above)
Comment (salogranada) [reply src/cv_camera_node.cpp]: Done
